src/main/java/com/yelp/nrtsearch/server/luceneserver/search/TerminateAfterWrapper.java

```diff
@@ -135,11 +134,18 @@ public void setScorer(Scorable scorer) throws IOException {
 
       @Override
       public void collect(int doc) throws IOException {
-        if (collectedDocCount.incrementAndGet() > terminateAfter) {
+        int counter = collectedDocCount.incrementAndGet();
+        if (counter > terminateAfter) {
           terminatedEarly = true;
-          throw new CollectionTerminatedException();
         }
-        leafCollector.collect(doc);
+
+        if (!terminatedEarly) {
+          // If we haven't terminated early, or if the collector is a TopHitsCollector, collect the doc
+          leafCollector.collect(doc);
+        } else if (leafCollector instanceof TotalHitCountCollector) {
+          // If we haven't terminated early, or if the collector is a TopHitsCollector, collect the doc
+          leafCollector.collect(doc);
```

Comment:
This is how lucene collects the totalHits:
https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/TotalHitCountCollector.java#L60